modules/apps/sharing/sharing-web/src/main/java/com/liferay/sharing/web/internal/display/SharingEntryPermissionDisplayAction.java

```diff
@@ -20,14 +20,14 @@
 public enum SharingEntryPermissionDisplayAction {
 
 	COMMENTS(
-		"COMMENTS", "comment", "collaborators-can-comment-on-the-item",
+		"COMMENTS", "comment", "collaborators-can-comment-on-the-item[",
```

Comment:
Oh!

This is the KEY of this. (thanks @jkappler for it)

if 'collaborators-can-comment-on-the-item[WHATEVER]' is not found it will use 'collaborators-can-comment-on-the-item' so 
-  Is this enum supposed to work with other types of models other than DLFileEntry? 
Yes, if not found for Blogs, it will be the simple 

If we desire that for anyClassName the new text is "hey mate, you will share something" we only need to add to the properties file
collaborators-can-comment-on-the-item[anyClassName]=hey mate, you will share something

and it will work!